src/main.cpp

```diff
@@ -21,5 +22,17 @@ int main(int argc, char** argv)
         return 1;
     }
 
-    return app.run(*mainWindow);
+    int retCode = 0;
+    if(Settings::instance().startAtTray())
+    {
+        mainWindow->hide();
+        Application::instance().keepAlive();
+        retCode = app.run();
```

Comment:
Does it work if you don't pass the mainwindow if you decide to show the app later from the tray icon?

Reply:
Yes. 

If i am not wrong, MainWindow::onShow adds window to application when called. `Application::instance().add_window(*this);`